### PR TITLE
Added Maven extensions to build the sample application in one reactor run.

### PR DESCRIPTION
**Pull request by lauritziu**
The pullrequest contains the following highlevel changes:

- provides a maven extension that makes it possible to package the p2 repository and the product in one maven module. There is no additional module necessary as in the sample at [e4-efxclipse-maven-sample](https://github.com/BestSolution-at/e4-efxclipse-maven-sample). This is done by deferring Tycho, which wants to verify the existence of the referenced generated p2-repo at a very early moment, where the repo was not yet genererated. Therefore the build fails. The extension enables tycho after the p2-repo was generated.
- provides a maven extension, which is able to generate all the maven dependency definitions for the target platform feature pom. All the information are extracted from the official efxclipse update site which is configurable. At the moment there is a whitelist used to limit the maven dependencies to the ones originally referenced in the e4-efxclipse-maven-sample. We would need to understand how the list in the sample was produced. 
- minor changes in the original plugins to improve the stability for projects having pom dependencies and error output
- all that is already in place and working in a patched sample (pullrequest will follow) [forked e4-efxclipse-maven-sample](https://github.com/lauritziu/e4-efxclipse-maven-sample)

It would be great, if we could talk about the target platform generation and the necessary dependencies. Our goal is to just give the update site url from efxclipse and you have the target platform setup.

### CODE REVIEW

**Comment by tomsontom** [comment]
Very cool! I'll look at itm later on. Just from the process point of view. Do you by chance have a signed ECA (https://wiki.eclipse.org/ECA). 

The reason I ask is that we maybe one day want to fold that work into e(fx)clipse (or some other Eclipse.org project) and it would make the process then easier because all contributions are coming from people who signed the ECA?

File: build/settings.xml [hunk 1]
```diff
@@ -0,0 +1,346 @@
+<settings>
```
Comment (tomsontom) [inline build/settings.xml]: I don't think this file should be in the pull request

Comment (lauritziu) [reply build/settings.xml]: Sure, I'll delete it.

File: maven-osgi-exec-plugin/src/main/java/at/bestsolution/maven/osgi/support/DebugSupport.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        CYAN("\u001B[36m"),
+        WHITE("\u001B[37m");
+
+        /**
+         * the end sequence to stop coloring
+         */
+        private final static String END_SEQUENCE = "\u001B[0m";
+
+        private TerminalOutputStyling(String escSequenz) {
+            this.escSequenz = escSequenz;
```
[the remark this answers is not in the document]
Comment (lauritziu) [reply maven-osgi-exec-plugin/src/main/java/at/bestsolution/maven/osgi/support/DebugSupport.java]: The namespace comes from the company I'm currently work for as freelancer. Would you like it to be at.bestsolution? I can even use my own namespace. No problem.

File: maven-osgi-package-plugin/src/main/java/at/bestsolution/maven/osgi/pack/DebugSupport.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        MAGENTA("\u001B[35m"),
+        CYAN("\u001B[36m"),
+        WHITE("\u001B[37m");
+
+        /**
+         * the end sequence to stop coloring
+         */
+        private final static String END_SEQUENCE = "\u001B[0m";
+
+        private TerminalOutputStyling(String escSequenz) {
```
[the remark this answers is not in the document]
Comment (lauritziu) [reply maven-osgi-package-plugin/src/main/java/at/bestsolution/maven/osgi/pack/DebugSupport.java]: Changed the wording in the last commit.

File: build/settings.xml (hunk 1 above)
Comment (lauritziu) [reply build/settings.xml]: Done.

Comment (lauritziu) [reply build/settings.xml]: I've just signed the ECA as user lauritziu_ii@yahoo.de. Is that enough. Do you need anything else?

File: maven-osgi-exec-plugin/src/main/java/at/bestsolution/maven/osgi/support/DebugSupport.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (tomsontom) [reply maven-osgi-exec-plugin/src/main/java/at/bestsolution/maven/osgi/support/DebugSupport.java]: yes if that is ok with your the company - then I'd prefer `at.bestsolution`. On the ECA: Yes this is fine! Let's document that stuff on the issue itself.

**Pull request merged**